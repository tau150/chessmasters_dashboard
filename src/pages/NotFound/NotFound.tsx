import { VStack, Box, Heading, Button, Text } from "@chakra-ui/react";
import { Link } from "react-router-dom";
import { ROUTES } from "@/routes/routes.types";

export const NotFound = () => {
  return (
    <VStack justify="center">
      <Heading color="orange.300">404!</Heading>
      <Heading>Something went wrong</Heading>
      <Text>The page you are looking for does not exist</Text>
      <Box mt="4">
        <Link to={ROUTES.ROOT}>
          <Button colorScheme="orange">Go to home</Button>
        </Link>
      </Box>
    </VStack>
  );
};
