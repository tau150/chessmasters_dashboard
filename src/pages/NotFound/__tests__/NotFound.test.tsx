import { expect } from "vitest";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { NotFound } from "../NotFound";

describe("NotFound page", () => {
  it("should match with snapshot", () => {
    const { container } = render(
      <BrowserRouter>
        <NotFound />
      </BrowserRouter>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
