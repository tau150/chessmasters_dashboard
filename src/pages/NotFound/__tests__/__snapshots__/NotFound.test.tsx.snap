// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotFound page > should match with snapshot 1`] = `
<div
  class="chakra-stack css-zliwli"
>
  <h2
    class="chakra-heading css-1jygyfq"
  >
    404!
  </h2>
  <h2
    class="chakra-heading css-0"
  >
    Something went wrong
  </h2>
  <p
    class="chakra-text css-0"
  >
    The page you are looking for does not exist
  </p>
  <div
    class="css-1fhgjcy"
  >
    <a
      href="/"
    >
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        Go to home
      </button>
    </a>
  </div>
</div>
`;
